### ISSUE TITLE
Boxes3D.from_tensor(mode="xyzwhd") cyclically permutes width, height, and depth

### ISSUE THREAD

**Issue by ducha-aiki** [kornia/kornia#4058]
`Boxes3D.from_tensor(..., mode="xyzwhd")` assigns the last three fields in the wrong order. The
semantic expected-value test uses equal dimensions (`3,3,3`), so the permutation is invisible; an
asymmetric gradcheck verifies derivatives rather than field semantics and also cannot catch it.

## Repro

```python
import torch
from kornia.geometry.boxes import Boxes3D

for dtype in (torch.float32, torch.float64):
    for shape in ((1, 6), (1, 1, 6)):
        value = torch.tensor([[10., 20., 30., 2., 3., 4.]], dtype=dtype).reshape(shape)
        out = Boxes3D.from_tensor(value, mode="xyzwhd").to_tensor(mode="xyzwhd")
        print(dtype, shape, value.flatten().tolist(), "->", out.flatten().tolist())
```

## Observed

Every dtype/rank cell prints:

```text
[10, 20, 30, 2, 3, 4] -> [10, 20, 30, 4, 2, 3]
```

The full sweep used three unequal dimension triples × float32/float64 × `(N,6)`/`(B,N,6)`:
12/12 map `(width,height,depth)` to `(depth,width,height)`. Source:

```python
depth, height, width = boxes[..., 4], boxes[..., 3], boxes[..., 5]
```

## Expected

`width, height, depth = boxes[...,3], boxes[...,4], boxes[...,5]`, and asymmetric round-trip tests
for both accepted ranks. This is a focused broken-output fix; it does not alter the mode contract.

## Classification

**Broken output**: all asymmetric `xyzwhd` inputs are assigned to the wrong axes. No current idiom
fits exactly; this is a field-order mapping defect.

## Disposition

n/a — single symbol.

## Environment

kornia 0.9.0rc1 @ `6fa75d64`, torch 2.9.1, Python 3.11.14, CPU. Found during bbox/boxes
conventions audit batch 4 (#4020); the audit does not change runtime behavior.

Posted on behalf of [@ducha-aiki](https://github.com/ducha-aiki) by Codex (GPT-5).

**Comment by Jeevang1-epic** [comment]
@ducha-aiki I'd like to take this one if it's still available.I went through the `Boxes3D.from_tensor(..., mode="xyzwhd")` path and I think I understand the issue. The width/height/depth fields are getting mapped in the wrong order, so asymmetric boxes expose the permutation pretty clearly.My plan is to reproduce it with a small asymmetric case first, add a focused round-trip/regression test, then fix the field mapping without changing the rest of the box contract. I'll also check both the batched and non-batched paths before opening the PR.

Looks pretty self-contained, so i should be able to work through this. If the scope looks right, could you assign it to me?

**Comment by ducha-aiki** [comment]
@Jeevang1-epic yes, it is yours — your plan is the right one, and the two things you called out (asymmetric repro first, both ranks) are exactly the two things that were missing. GitHub does not let me put a non-collaborator in the assignee field, so this comment is the claim marker; nobody else should pick it up.

Three things I checked on `main` @ `ee6b713f` so you do not have to re-derive them:

**1. Only the input side is wrong — the fix is one line.** `to_tensor("xyzwhd")` is already correct, so do not touch it:

```
xyzxyz [10,20,30,12,23,34] (w=2,h=3,d=4) -> to_tensor("xyzwhd") = [10,20,30,2,3,4]   correct
from_tensor(xyzwhd [10,20,30,2,3,4]) -> to_tensor("xyzxyz")    = [10,20,30,14,22,33] wrong
```

`14,22,33` is `x+d, y+w, z+h`, which localizes the permutation entirely to `kornia/geometry/boxes.py:1141-1142`:

```python
elif mode == "xyzwhd":
    depth, height, width = boxes[..., 4], boxes[..., 3], boxes[..., 5]
```

**2. Do not "fix" `Boxes3D.get_boxes_shape` (`boxes.py:1061`) while you are in the file.** It returns `(depths, heights, widths)` — that order is deliberate and documented in its own docstring and doctest, and it reads `to_tensor("xyzwhd")` correctly. It looks like the same bug and is not.

**3. Blast radius is user-facing only.** `grep -rn 'xyzwhd' kornia/` outside `boxes.py` is empty — nothing inside the library builds a `Boxes3D` in that mode, so there is no compensating call site to repair alongside the fix.

For the tests, the reason this survived is `tests/geometry/test_boxes.py:744`, which uses `[1, 2, 3, 3, 3, 3]` — `w == h == d == 3`, so the permutation is invisible. Make that case asymmetric rather than adding a new test beside it and leaving a symmetric one in place. The `gradcheck` at `:1010` passes on both revisions; it is not coverage for this. Please cover both accepted ranks, `(N, 6)` and `(B, N, 6)`.

Before you open the PR: run your new test with `kornia/` swapped to the merge-base and paste the fail-on-base output in the PR body, and add a `## Unreleased` → `Bug fixes` entry in `CHANGELOG.md` citing your PR number.

Posted on behalf of @ducha-aiki by Claude (Opus 5).

**Comment by ducha-aiki** [comment]
Retro-tagged: the classification now names **idiom 7** (axis-order slip hidden by symmetric test data), ratified 2026-09-10 with #4269 as its second instance. Nothing else in this closed issue changes.

Posted on behalf of @ducha-aiki by Claude (Fable 5.1).
